### PR TITLE
Additional code cleanup

### LINKED ISSUE #35164: Additional followup work to cleanup unnecessary GC config value access in eeconfig.cpp
This issue is filed to follow up on [this issue comment](https://github.com/dotnet/runtime/pull/34797#issuecomment-612203252)

### PR DESCRIPTION
**Pull request by cshung**
Fixes #35164
This should cover all configuration value that is not used anywhere other than gcenv.ee.cpp

### CODE REVIEW

File: src/coreclr/src/vm/eeconfig.h [hunk 1]
```diff
@@ -684,25 +669,13 @@ class EEConfig
     int  iGCStress;
 #endif
 
-#define DEFAULT_GC_PRN_LVL 3
-    size_t iGCgen0size;
-    size_t iGCSegmentSize;
+#define DEFAULT_GC_PRN_LVL 3
```
Comment (Maoni0) [inline src/coreclr/src/vm/eeconfig.h:672]: you can get rid of this define too

File: src/coreclr/src/vm/eeconfig.cpp [hunk 2]
```diff
@@ -560,23 +551,6 @@ fTrackDynamicMethodDebugInfo = CLRConfig::GetConfigValue(CLRConfig::UNSUPPORTED_
 
 #endif //STRESS_HEAP
 
-#ifdef HOST_64BIT
```
Comment (Maoni0) [inline src/coreclr/src/vm/eeconfig.cpp:563]: we should get rid of these in clrconfigvalues.h too

Comment (cshung) [reply src/coreclr/src/vm/eeconfig.cpp:563]: That would conflict with https://github.com/dotnet/runtime/pull/35259. I will do that once that is merged.

**Review by Maoni0 — APPROVED** [review]

File: src/coreclr/src/vm/eeconfig.h [hunk 3]
```diff
@@ -401,28 +401,13 @@ class EEConfig
...
 
-    size_t  GetGCgen0size  ()               const {LIMITED_METHOD_CONTRACT; return iGCgen0size;   }
-    void    SetGCgen0size  (size_t iSize)         {LIMITED_METHOD_CONTRACT; iGCgen0size = iSize;   }
-    size_t  GetSegmentSize ()               const {LIMITED_METHOD_CONTRACT; return iGCSegmentSize; }
-    void    SetSegmentSize (size_t iSize)         {LIMITED_METHOD_CONTRACT; iGCSegmentSize = iSize; }
-
     int     GetGCconcurrent()               const {LIMITED_METHOD_CONTRACT; return iGCconcurrent; }
     void    SetGCconcurrent(int val)              {LIMITED_METHOD_CONTRACT; iGCconcurrent = val;  }
 #ifdef _DEBUG
     int     GetGCLatencyMode()              const {LIMITED_METHOD_CONTRACT; return iGCLatencyMode; }
```
Comment (jkotas) [inline src/coreclr/src/vm/eeconfig.h:407]: Is this `GetGCLatencyMode` used anywhere?

File: src/coreclr/src/inc/clrconfigvalues.h [hunk 4]
```diff
@@ -271,12 +271,9 @@ RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_gcConservative, W("gcConservative"), 0, "En
 RETAIL_CONFIG_DWORD_INFO(UNSUPPORTED_gcServer, W("gcServer"), 0, "Enables server GC")
 CONFIG_STRING_INFO(INTERNAL_GcCoverage, W("GcCoverage"), "Specify a method or regular expression of method names to run with GCStress")
 CONFIG_STRING_INFO(INTERNAL_SkipGCCoverage, W("SkipGcCoverage"), "Specify a list of assembly names to skip with GC Coverage")
-RETAIL_CONFIG_DWORD_INFO_DIRECT_ACCESS(UNSUPPORTED_gcForceCompact, W("gcForceCompact"), "When set to true, always do compacting GC")
-RETAIL_CONFIG_DWORD_INFO_DIRECT_ACCESS(UNSUPPORTED_GCgen0size, W("GCgen0size"), "Specifies the smallest gen0 size")
 RETAIL_CONFIG_DWORD_INFO_DIRECT_ACCESS(UNSUPPORTED_GCGen0MaxBudget, W("GCGen0MaxBudget"), "Specifies the largest gen0 allocation budget")
```
Comment (jkotas) [inline src/coreclr/src/inc/clrconfigvalues.h:274]: More nits: Number of these have been moved to the GC and can be deleted from here.

**Review by jkotas — APPROVED** [review]

**Pull request merged**